### PR TITLE
Upmerge 19.06.2023

### PR DESCRIPTION
**Pull request by rlubos**
Manual selection of integration branches:
test-sdk-nrf: sdk-nrf-PR-11559
test-ble: refs/pull-requests/4216/from

### CODE REVIEW

File: samples/matter/lock/Kconfig [hunk 1]
```diff
@@ -97,6 +97,14 @@ endif # CHIP_WIFI
 config CHIP_ICD_SUBSCRIPTION_HANDLING
 	default y
 
+config MATTER_LOCK_LTO
```
Comment (Damian-Nordic) [inline samples/matter/lock/Kconfig:100]: Please do not include this change. It was attempted as a workaround by @awojasinski-nordicsemi to fix the flash overflow, but apparently it didn't help and the flash overflow was fixed by other means.

Comment (rlubos) [reply samples/matter/lock/Kconfig:100]: ok, I'll drop this commit

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -177,6 +177,17 @@ nRF Desktop
     This is done to avoid flooding logs during USB state changes.
   * If the USB state is set to :c:enum:`USB_STATE_POWERED`, the :ref:`nrf_desktop_usb_state_pm` restricts the power down level to the :c:enum:`POWER_MANAGER_LEVEL_SUSPENDED` instead of requiring :c:enum:`POWER_MANAGER_LEVEL_ALIVE`.
     This is done to prevent the device from powering down and waking up multiple times when an USB cable is connected.
+  * The :ref:`nrf_desktop_dfu` automatically enables 8-bit write block size emulation (:kconfig:option:`CONFIG_SOC_FLASH_NRF_EMULATE_ONE_BYTE_WRITE_ACCESS`) to ensure that update images with sizes not aligned to word size can be successfully stored in the internal flash.
```
Comment (MarekPieta) [inline doc/nrf/releases/release-notes-changelog.rst:180]: It seems that the cherry-pick contains more than it should ;)

**Review by MarekPieta — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-changelog.rst (hunk 2 above)
Comment (rlubos) [reply doc/nrf/releases/release-notes-changelog.rst:180]: What do you mean exactly, I just took it from https://github.com/nrfconnect/sdk-nrf/pull/11057/commits - what should be removed?

Comment (MarekPieta) [reply doc/nrf/releases/release-notes-changelog.rst:180]: Ahh I see... Seems like rebase issue happened there. Only the
```
* Disabled ``CONFIG_BOOT_SERIAL_IMG_GRP_HASH`` in MCUboot bootloader release configurations of boards that use nRF52820 SoC.
    This is done to reduce the memory consumption.
```
release note should be added here (remaining release notes are related to previous release).

File: samples/matter/lock/Kconfig (hunk 1 above)
Comment (awojasinski-nordicsemi) [reply samples/matter/lock/Kconfig]: Yup, I can confirm that it should be dropped

File: doc/nrf/releases/release-notes-changelog.rst (hunk 2 above)
Comment (rlubos) [reply doc/nrf/releases/release-notes-changelog.rst]: Ok, that makes sense, fixed the changelog.

File: samples/nrf9160/lwm2m_client/src/sensors/light_sensor.c [hunk 3]
```diff
@@ -38,7 +38,7 @@ static const struct device *light_sensor_dev = DEVICE_DT_GET(DT_NODELABEL(bh1749
 static int64_t fetch_timestamp;
 
 
-static int sensor_read(struct sensor_value measurements[])
+static int _sensor_read(struct sensor_value measurements[])
```
Comment (glarsennordic) [inline samples/nrf9160/lwm2m_client/src/sensors/light_sensor.c:41]: Maybe this should be given a more specific name? Like `light_sensor_read` or something?

**Review by glarsennordic — APPROVED** [review]

File: samples/nrf9160/lwm2m_client/src/sensors/light_sensor.c (hunk 3 above)
Comment (rlubos) [reply samples/nrf9160/lwm2m_client/src/sensors/light_sensor.c:41]: The thing is, this function is not specific. There already is `light_sensor_read()` in this file, along with `color_sensor_read()`, and this helper func is called from both. I'm open for other name suggestions though.

Comment (MirkoCovizzi) [reply samples/nrf9160/lwm2m_client/src/sensors/light_sensor.c:41]: This is a common and quick way to fix a name conflict and it is used elsewhere too. I think that it is fine for an upmerge.

Comment (glarsennordic) [reply samples/nrf9160/lwm2m_client/src/sensors/light_sensor.c:41]: Oh my bad, I hadn't noticed those other naming conflicts. I don't have a better alternative then, I'll resolve

**Review by sachinthegreen — APPROVED** [review]

**Comment by tomchy** [comment]
@sebastiandraus : Could you schedule Zigbee NCP tests on this PR?
The NCP sample is the only one that uses the `NRF_STORE_REBOOT_TYPE_GPREGRET` functionality.

**Comment by tomchy** [comment]
@rlubos I have no idea what's wrong with GitHub today. I tried to add Sebastian to the review and it says that I removed a bunch of people. I hope that any commit will restore the list of reviewers, based on the code owners entries.

**Review by sebastiandraus — APPROVED** [review]

**Comment by sebastiandraus** [comment]
@tomchy ,
> @sebastiandraus : Could you schedule Zigbee NCP tests on this PR? The NCP sample is the only one that uses the `NRF_STORE_REBOOT_TYPE_GPREGRET` functionality.

NCP is is fixed, verified by full nightly run

**Review by SeppoTakalo — APPROVED** [review]

**Review by tomchy — APPROVED** [review]

**Review by ludvigsj — APPROVED** [review]

**Review by pdunaj — APPROVED** [review]

**Review by gregersrygg — APPROVED** [review]

**Review by sigvartmh — APPROVED** [review]

**Comment by rlubos** [comment]
All jobs passed after reruns, merging.

**Pull request merged**